appyx-components/backstack/common/src/commonMain/kotlin/com/bumble/appyx/components/backstack/ui/parallax/BackStackParallax.kt

```diff
@@ -0,0 +1,97 @@
+import com.bumble.appyx.interactions.core.ui.context.UiContext
+import com.bumble.appyx.interactions.core.ui.gesture.Drag
+import com.bumble.appyx.interactions.core.ui.gesture.Gesture
+import com.bumble.appyx.interactions.core.ui.gesture.GestureFactory
+import com.bumble.appyx.interactions.core.ui.gesture.dragHorizontalDirection
+import com.bumble.appyx.interactions.core.ui.state.MatchedTargetUiState
+import com.bumble.appyx.transitionmodel.BaseMotionController
+
+class BackstackParallax<InteractionTarget : Any>(
+
+    uiContext: UiContext,
+    defaultAnimationSpec: SpringSpec<Float> = DefaultAnimationSpec
+) : BaseMotionController<InteractionTarget, State<InteractionTarget>, MutableUiState, TargetUiState>(
+    uiContext = uiContext,
+    defaultAnimationSpec = defaultAnimationSpec,
+) {
+    private val width = uiContext.transitionBounds.widthDp.value
+
+    private val left = TargetUiState(
+        elementWidth = width,
+        offsetMultiplier = -1f,
+    )
+    private val right = TargetUiState(
+        elementWidth = width,
+        offsetMultiplier = 1f,
+    )
+
+    private val bottom = TargetUiState(
+        elementWidth = width,
+        offsetMultiplier = -0.2f,
+    )
+
+    private val top = TargetUiState(
+        elementWidth = width,
+        offsetMultiplier = 0f,
+    )
+
+    override fun State<InteractionTarget>.toUiTargets(): List<MatchedTargetUiState<InteractionTarget, TargetUiState>> {
+        val stashed = stashed.mapIndexed { index, element ->
+            MatchedTargetUiState(
+                element = element,
+                targetUiState = if (index == stashed.size - 1) bottom else left,
+            )
+        }
+
+        return stashed + listOf(
+            MatchedTargetUiState(
+                element = active,
+                targetUiState = top
+            )
+        ) + (created + destroyed).map {
+            MatchedTargetUiState(
+                element = it,
+                targetUiState = right
+            )
+        }
+    }
+
+    override fun mutableUiStateFor(
+        uiContext: UiContext,
+        targetUiState: TargetUiState
+    ): MutableUiState =
+        targetUiState.toMutableState(uiContext)
+
+    class Gestures<InteractionTarget : Any>(
+        transitionBounds: TransitionBounds,
+    ) : GestureFactory<InteractionTarget, State<InteractionTarget>> {
+        private val width = transitionBounds.screenWidthDp
+
+        override fun createGesture(
+            delta: Offset,
+            density: Density
+        ): Gesture<InteractionTarget, State<InteractionTarget>> {
+
+            val widthRight = with(density) { width.toPx() }
+
+            return if  (dragHorizontalDirection(delta) == Drag.HorizontalDirection.RIGHT) {
+                Gesture(
+                    operation = Pop(),
+                    dragToProgress = { offset -> (offset.x / widthRight) },
+                    partial = { offset, partial -> offset.copy(x = partial * widthRight) },
```

Comment:
In this [PR](https://github.com/bumble-tech/appyx/pull/455), @zsoltk introduced a really cool simplification that alleviates us to implement `dragToProgress` and `partial` methods. I suggest you to take a look at it, for instance this gesture handler for dating cards can help you with it: 
https://github.com/bumble-tech/appyx/pull/455/files#diff-12de37ea06c670807f7b67dbb9632aeaa26f27364b7335a06118cc317354d18c

Reply:
not only that, but this API is now internal, so after you merge main it will throw a compilation error

Reply:
Oh great! Thanks for the suggestion.